controllers/dependencies.go

```diff
@@ -46,22 +41,22 @@ func applyExternalArtifactsDownload(ispn *infinispanv1.Infinispan, spec *corev1.
 	volumeMounts := &ispnContainer.VolumeMounts
 	containerPosition := kube.ContainerIndex(*initContainers, ExternalArtifactsDownloadInitContainer)
 	if ispn.HasExternalArtifacts() {
-		extractCommands, err := externalArtifactsExtractCommand(ispn)
-		if err != nil {
-			retErr = err
-			return
-		}
+		serverLibs := serverLibs(ispn)
 		if containerPosition >= 0 {
-			if spec.InitContainers[containerPosition].Args[0] != extractCommands {
-				spec.InitContainers[containerPosition].Args = []string{extractCommands}
+			if spec.InitContainers[containerPosition].Env[0].Value != serverLibs {
+				spec.InitContainers[containerPosition].Env[0].Value = serverLibs
 				updated = true
 			}
 		} else {
 			*initContainers = append(*initContainers, corev1.Container{
-				Image:   ispn.ImageName(),
-				Name:    ExternalArtifactsDownloadInitContainer,
-				Command: []string{"sh", "-c"},
-				Args:    []string{extractCommands},
+				Image: ispn.ImageName(),
+				Name:  ExternalArtifactsDownloadInitContainer,
+				Env: []corev1.EnvVar{
+					{Name: "SERVER_LIBS", Value: serverLibs},
+					{Name: "SERVER_LIBS_DIR", Value: "server/lib/external-artifacts"},
```

Comment:
How about a const used here and for the ExternalActifactsMountPath ?

Reply:
Fixed.